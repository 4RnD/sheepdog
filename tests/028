#!/bin/bash

# Test sheep recovery logic

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup

for i in `seq 0 1`; do _start_sheep $i; done

_wait_for_sheep 2

$COLLIE cluster format -c 2

$COLLIE vdi create test 20M

dd if=/dev/urandom | $COLLIE vdi write test
$COLLIE vdi read test | md5sum > /tmp/csum.0

for i in 2 3; do
    _start_sheep $i
    sleep 1
    _kill_sheep $(($i-2))
    _wait_for_sheep_recovery $i
done

dd if=/dev/urandom | $COLLIE vdi write test -p 7002
$COLLIE vdi read test -p 7002 | md5sum > /tmp/csum.1

for i in 2 3; do
    _start_sheep $(($i-2))
    sleep 1
    _kill_sheep $i
    _wait_for_sheep_recovery $(($i-2))
done

$COLLIE vdi read test | md5sum > /tmp/csum.2
$COLLIE vdi read test -p 7001 | md5sum > /tmp/csum.3

$COLLIE vdi object test

for i in `seq 0 4`; do
    $COLLIE vdi object -i $i test
done

if [ "$(diff -u /tmp/csum.0 /tmp/csum.1)" ]; then
    echo "update test successfully"
fi

diff -u /tmp/csum.1 /tmp/csum.2
diff -u /tmp/csum.1 /tmp/csum.3
