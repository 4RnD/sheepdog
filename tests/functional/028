#!/bin/bash

# Test sheep recovery logic

. ./common

for i in `seq 0 1`; do _start_sheep $i; done

_wait_for_sheep 2

_cluster_format -c 2

$COLLIE vdi create test 100M

for i in `seq 0 24`; do
    echo $i | $COLLIE vdi write test $((i * 4 * 1024 * 1024)) 512
done

$COLLIE vdi read test | md5sum

for i in 2 3; do
    _start_sheep $i
    sleep 1
    _kill_sheep $(($i-2))
    _wait_for_sheep_recovery $i
done

# write different content
for i in `seq 0 24`; do
    echo $(($i+1)) | $COLLIE vdi write test $((i * 4 * 1024 * 1024)) 512 -p 7002
done

$COLLIE vdi read test -p 7002 | md5sum

for i in 2 3; do
    _start_sheep $(($i-2))
    sleep 1
    _kill_sheep $i
    _wait_for_sheep_recovery $(($i-2))
done

$COLLIE vdi read test | md5sum
$COLLIE vdi read test -p 7001 | md5sum

$COLLIE vdi object test

for i in `seq 0 24`; do
    $COLLIE vdi object -i $i test
done
