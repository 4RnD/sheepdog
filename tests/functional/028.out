QA output created by 028
using backend plain store
9c7766570b3be3aff2724f587c2f4107  -
a6c5d562057d0b8e7c90df17e5ea9450  -
a6c5d562057d0b8e7c90df17e5ea9450  -
a6c5d562057d0b8e7c90df17e5ea9450  -
Looking for the inode object 0x7c2b25 with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000000 (vid 0x7c2b25 idx 0, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000001 (vid 0x7c2b25 idx 1, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000002 (vid 0x7c2b25 idx 2, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000003 (vid 0x7c2b25 idx 3, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000004 (vid 0x7c2b25 idx 4, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000005 (vid 0x7c2b25 idx 5, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000006 (vid 0x7c2b25 idx 6, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000007 (vid 0x7c2b25 idx 7, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000008 (vid 0x7c2b25 idx 8, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000009 (vid 0x7c2b25 idx 9, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b250000000a (vid 0x7c2b25 idx 10, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b250000000b (vid 0x7c2b25 idx 11, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b250000000c (vid 0x7c2b25 idx 12, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b250000000d (vid 0x7c2b25 idx 13, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b250000000e (vid 0x7c2b25 idx 14, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b250000000f (vid 0x7c2b25 idx 15, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000010 (vid 0x7c2b25 idx 16, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000011 (vid 0x7c2b25 idx 17, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000012 (vid 0x7c2b25 idx 18, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000013 (vid 0x7c2b25 idx 19, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000014 (vid 0x7c2b25 idx 20, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000015 (vid 0x7c2b25 idx 21, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000016 (vid 0x7c2b25 idx 22, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7000 127.0.0.1:7001
Looking for the object 0x7c2b2500000017 (vid 0x7c2b25 idx 23, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
Looking for the object 0x7c2b2500000018 (vid 0x7c2b25 idx 24, 2 copies) with 2 nodes

127.0.0.1:7000 has the object
127.0.0.1:7001 has the object

According to sheepdog algorithm, the object should be located at:
127.0.0.1:7001 127.0.0.1:7000
